frontend/components/main.js

```diff
@@ -139,12 +140,6 @@ export default class Main extends React.Component {
           {this.renderCategory(category, definitions)}
         </div>
       )
-    } else if (categoryId) {
```

Comment:
What sort of circumstances (user action) would have gotten us here with the old implementation? I think this happened if I added a random/non-existent category to the url in the browser. If I do that now (say http://shields-staging-pr-2906.herokuapp.com/category/foo) I'm getting the 404 page. Is that what we want to do now vs. the unknown category we used to show (https://shields-staging.herokuapp.com/#/examples/foo)

Reply:
The old route would match anything that was typed in to this page. You can see the code that did that here: https://github.com/badges/shields/pull/2906/files#diff-e14ec8fd2b038fbccea5f8090d26ace4L11

That means it was possible for this component to be loaded with an invalid category.

Now the routes are established by the `createPage()` calls in `gatsby-node.js` so we don't reach this point if the category doesn't exist.

If we want a custom 404 page that's probably possible, though it would no longer be the responsibility of this component.

Reply:
I think the new behavior is good (and I'm a fan of the existing 404 page anyway). If I manually type in a path in the url that doesn't exist, i deserve a 404 IMO 👍